// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option objc_class_prefix = "AFS";

package wiscAFS;

// The greeting service definition.
service AFSController {
  // Sends a greeting
  rpc OpenFile (RPCRequest) returns (RPCResponse) {}
  rpc CloseFile (RPCRequest) returns (RPCResponse) {}
  rpc CreateFile (RPCRequest) returns (RPCResponse) {}
  rpc DeleteFile (RPCRequest) returns (RPCResponse) {}
  rpc CreateDir (RPCRequest) returns (RPCResponse) {}
  rpc RemoveDir (RPCRequest) returns (RPCResponse) {}
  rpc GetAttr (RPCRequest) returns (RPCResponse) {}
  rpc OpenDir (RPCRequest) returns (RPCResponse) {}
  rpc ReadDir (RPCRequest) returns (RPCResponse) {}

}

// The request message containing the user's name.
message RPCRequest {
  string filename = 1;
  string path = 2;
  int32 mode = 3;
  int32 flags = 4;
  string data = 5;
  int32 fileDescriptor = 6;
}

message FileInfo {
    int32 st_dev = 1;
    int32 st_ino = 2;
    int32 st_mode = 3;
    int32 st_nlink = 4;
    int32 st_uid = 5;
    int32 st_gid = 6;
    int32 st_rdev = 7;
    int64 st_size = 8;
    int64 st_blksize = 9;
    int64 st_blocks = 10;
    int64 st_atim = 11;
    int64 st_mtim = 12;
    int64 st_ctim = 13;
}

message RPCResponse {
  int32 status = 1;
  string data = 2;
  optional FileInfo fileInfo = 4;

}
